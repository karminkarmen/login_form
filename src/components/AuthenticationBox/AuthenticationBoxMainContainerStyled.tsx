import styled from 'styled-components';
import { centered } from '../../configs/mixins';

const AuthenticationBoxMainContainerStyled = styled.div`
    ${centered}
    flex-direction: column;
    border-radius:  ${props => props.theme.radius.maximum};
    background: ${props => props.theme.colors.secondary};
    width: ${props => props.theme.width.mainElementWidth};
    height: ${props => props.theme.width.mainElementWidth};
`;

export default AuthenticationBoxMainContainerStyled;
