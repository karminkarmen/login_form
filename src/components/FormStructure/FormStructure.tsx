import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import FormContainerStyled from './FormStructureStyled';
import FieldsetStyled from '../common/FieldsetStyled';
import LoginButton from '../LoginButton/LoginButton';
import TextInput from '../TextInput/TextInput';
import { loginStatuses } from '../../constants/statusesConsts';
import ParagraphStyled from '../common/ParagraphStyled';
import CheckboxInput from '../CheckboxInput/CheckboxInput';

type Props = {
    formSubmitHandler: (e: any) => void, 
    onEmailChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
    onPasswordChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
    emailValidationInfo: string
    passwordValidationInfo: string,
    loginStatus: loginStatuses,
}

const FormStructure: React.FC<Props> = ({ 
    formSubmitHandler, 
    onEmailChange, 
    emailValidationInfo, 
    onPasswordChange, 
    passwordValidationInfo, 
    loginStatus 
}) => {
    return (
        <ThemeProvider theme={theme}>
            <form 
                method="POST" 
                action="" 
                onSubmit={formSubmitHandler} 
            >
                <FieldsetStyled>
                    <FormContainerStyled theme={theme}>
                        <TextInput 
                            handler={onEmailChange} 
                            label={'email'} 
                            type={'text'} 
                            validationInfo={emailValidationInfo} 
                        />
                        <TextInput 
                            handler={onPasswordChange} 
                            label={'password'} 
                            type={'password'} 
                            validationInfo={passwordValidationInfo} 
                        />
                        <CheckboxInput 
                            name={'remember'}
                            label={'Remember me'}
                        />
                        { loginStatus === loginStatuses.REJECTED && <ParagraphStyled>Invalid email or password</ParagraphStyled>}
                    </FormContainerStyled>
                    <LoginButton />
                </FieldsetStyled>
            </form> 
        </ThemeProvider>
    );
};

export default FormStructure;


