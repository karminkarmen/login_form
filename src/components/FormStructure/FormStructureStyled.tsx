import styled from 'styled-components';
import { centered } from '../../configs/mixins';

const FormStructureStyled = styled.div`
    flex-direction: column;
    ${centered}
    border-radius:  ${props => props.theme.radius.maximum};
    width: ${props => props.theme.width.mainElementWidth};
    height: ${props => props.theme.width.mainElementWidth};
`;

export default FormStructureStyled;