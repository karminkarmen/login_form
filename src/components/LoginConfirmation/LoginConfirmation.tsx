import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { Tick } from '@styled-icons/typicons';
import theme from '../../configs/theme';
import ParagraphStyled from '../common/ParagraphStyled';
import { innerWidthSize } from '../../constants/themeConsts';

const StyledTick = styled(Tick)`
    color: ${props => props.theme.colors.main};
    height: ${innerWidthSize / 2}px;
`
const LoginConfirmation: React.FC = () => (
    <ThemeProvider theme={theme}>
        <StyledTick />
        <ParagraphStyled bigFontSize textSpace>login successful</ParagraphStyled>
    </ThemeProvider>
);

export default LoginConfirmation;
