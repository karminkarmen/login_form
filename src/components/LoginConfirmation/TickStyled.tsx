import styled from 'styled-components';
import { Tick } from '@styled-icons/typicons';
import { innerWidthSize } from '../../constants/themeConsts';

export const TickStyled = styled(Tick)`
    color: ${props => props.theme.colors.main};
    height: ${innerWidthSize / 2}px;
`
export default TickStyled;
