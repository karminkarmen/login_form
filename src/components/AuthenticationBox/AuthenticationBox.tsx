import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import FormStructure from '../FormStructure/FormStructure';
import AuthenticationBoxMainContainerStyled from './AuthenticationBoxMainContainerStyled';
import LoginConfirmation from '../LoginConfirmation/LoginConfirmation';
import { formValidator } from '../../validation/formValidator';
import loginHandler from '../../apiMock/loginHandler';
import { LoginStatus } from '../../constants/statusesConsts';

const AuthenticationBox: React.FC = () => {
    const [password, setPassword] = React.useState('');
    const [email, setEmail] = React.useState('');

    const [emailValidationInfo, setEmailValidationInfo] = React.useState('');
    const [passwordValidationInfo, setPasswordValidationInfo] = React.useState('');
    const [loginStatus, setLoginStatus] = React.useState(LoginStatus.NOT_LOGGED_IN);

    const { validateEmail, validatePassword } = formValidator();

    const formSubmitHandler = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();

        const emailValidationOutput = validateEmail(email);
        const passwordValidationOutput = validatePassword(password);

        setPasswordValidationInfo(passwordValidationOutput);
        setEmailValidationInfo(emailValidationOutput);

        if (emailValidationOutput.length === 0 && passwordValidationOutput.length === 0) {
            loginHandler(email, password)
                .then((res: unknown | PromiseLike<void>) => setLoginStatus(LoginStatus.AUTHORIZED))
                .catch(e => setLoginStatus(LoginStatus.REJECTED));
        }
    };

    const onEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => setEmail(e.target.value);
    const onPasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => setPassword(e.target.value);

    return (
        <ThemeProvider theme={theme}>
            <AuthenticationBoxMainContainerStyled>
                { (loginStatus === LoginStatus.REJECTED || loginStatus === LoginStatus.NOT_LOGGED_IN) && 
                    <FormStructure 
                        formSubmitHandler={formSubmitHandler}
                        onEmailChange={onEmailChange} 
                        onPasswordChange={onPasswordChange} 
                        emailValidationInfo={emailValidationInfo} 
                        passwordValidationInfo={passwordValidationInfo}
                        loginStatus={loginStatus}
                    />
                }
                {(loginStatus === LoginStatus.AUTHORIZED) && <LoginConfirmation />}
            </AuthenticationBoxMainContainerStyled>
        </ThemeProvider>
    );
};

export default AuthenticationBox;
