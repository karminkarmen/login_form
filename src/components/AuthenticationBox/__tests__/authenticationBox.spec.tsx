import React from 'react';
import { shallow } from 'enzyme';

import AuthenticationBox from '../AuthenticationBox';
import FormStructure from '../../FormStructure/FormStructure';
import LoginConfirmation from '../../LoginConfirmation/LoginConfirmation';
import AuthenticationBoxMainContainerStyled from '../AuthenticationBoxMainContainerStyled';

describe('<AuthenticationBox />', () => {

    it('matches snapshot', () => {
        const wrapper = shallow(<AuthenticationBox />);

        console.log(wrapper.dive().html())
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should not display confirmation panel if user is not logged in', () => {
        const wrapper = shallow(<AuthenticationBox />);

        expect(wrapper.find(LoginConfirmation)).toHaveLength(0);
    });

    it('should display form container', () => {
        const wrapper = shallow(<AuthenticationBox />);

        expect(wrapper.find(AuthenticationBoxMainContainerStyled)).toHaveLength(1);
    });

    it('should have proper initial props for form structure', () => {
        const wrapper = shallow(<AuthenticationBox />);

        expect(wrapper.find(FormStructure).props()).toEqual(
            {
                formSubmitHandler: expect.any(Function),
                onEmailChange: expect.any(Function),
                onPasswordChange: expect.any(Function),
                emailValidationInfo: '',
                passwordValidationInfo: '',
                loginStatus: 'NOT_LOGGED_IN'
            }
        )
    });

    

});
