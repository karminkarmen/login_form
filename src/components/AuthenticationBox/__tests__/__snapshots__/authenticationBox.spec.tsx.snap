// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticationBox /> matches snapshot 1`] = `
"<ThemeProvider theme={{...}}>
  <styled.div>
    <FormStructure formSubmitHandler={[Function: formSubmitHandler]} onEmailChange={[Function: onEmailChange]} onPasswordChange={[Function: onPasswordChange]} emailValidationInfo=\\"\\" passwordValidationInfo=\\"\\" loginStatus=\\"NOT_LOGGED_IN\\" />
  </styled.div>
</ThemeProvider>"
`;
